helper/identity/templating.go

```diff
@@ -196,19 +262,91 @@ func performTemplating(ns *namespace.Namespace, input string, entity *Entity, gr
 		return "", ErrTemplateValueNotFound
 	}
 
+	performTimeTemplating := func(trimmed string) (string, error) {
+		opsSplit := strings.SplitN(trimmed, ".", 3)
+
+		if opsSplit[0] != "now" {
+			return "", fmt.Errorf("invalid time selector %q", opsSplit[0])
+		}
+
+		result := time.Now()
+		switch len(opsSplit) {
+		case 1:
+			// return current time
+		case 2:
+			return "", errors.New("missing time operand")
+		case 3:
+			duration, err := time.ParseDuration(opsSplit[2])
+			if err != nil {
+				return "", errwrap.Wrapf("invalid duration: {{err}}", err)
+			}
+
+			switch opsSplit[1] {
+			case "plus":
+				result = result.Add(duration)
+			case "minus":
+				result = result.Add(-duration)
+			default:
+				return "", fmt.Errorf("invalid time operator %q", opsSplit[1])
+			}
+		}
+
+		return strconv.FormatInt(result.Unix(), 10), nil
+	}
+
 	switch {
 	case strings.HasPrefix(input, "identity.entity."):
-		if entity == nil {
+		if p.Entity == nil {
 			return "", ErrNoEntityAttachedToToken
 		}
 		return performEntityTemplating(strings.TrimPrefix(input, "identity.entity."))
 
 	case strings.HasPrefix(input, "identity.groups."):
-		if len(groups) == 0 {
+		if len(p.Groups) == 0 {
 			return "", ErrNoGroupsAttachedToToken
 		}
 		return performGroupsTemplating(strings.TrimPrefix(input, "identity.groups."))
+
+	case strings.HasPrefix(input, "time."):
+		return performTimeTemplating(strings.TrimPrefix(input, "time."))
 	}
 
 	return "", ErrTemplateValueNotFound
 }
+
+func listGroups(groups []*Group, element string) string {
```

Comment:
I would make this two funcs, or add checking for invalid `element`.

Reply:
Split into member-specific funcs